packages/graphql_server/src/graphql/resolver/bookmark.ts

```diff
@@ -0,0 +1,23 @@
+import { bookmarkIsAlreadyInDB, insertBookmark } from '../../supabaseUtils'
+
+/**
+ * Bookmarks a projects for a user.
+ * @param {string} userID - The user ID of the user in question.
+ * @param {string} projectID - The project ID of the project in question.
+ * @param {string} category - The category the bookmark should be added to.
+ */
+export const addBookmark = async (userID: string, projectID: string, category: string) => {
+  if (await bookmarkIsAlreadyInDB(userID, projectID)) {
```

Comment:
upsert would be better

Reply:
I don't think it would. 
lets say a user has a project bookmarked in category "june". He forgot so he wants to add the same bookmark again the next day. He wants to add it to category "july" upserting would cahnge the category and return a success code. 
But I think it is better if it returns that the project is already bookmarked